### ISSUE TITLE
New Sonos provider - announcement error:  'SonosPlayer' object has no attribute 'zone_name'

### ISSUE THREAD

**Issue by jordandrako** [music-assistant/hass-music-assistant#2803]
### What version of Music Assistant has the issue?

2.3.0b3

### What version of the Home Assistant Integration have you got installed?

2024.8.1

### Have you tried everything in the Troubleshooting FAQ and reviewed the Open and Closed Issues and Discussions to resolve this yourself?

- [X] Yes

### The problem

When using the `announce: true` option on the `media_player.play_media`, I now get this error in the script trace: `Error: MusicAssistantError: 'SonosPlayer' object has no attribute 'zone_name'`

![image](https://github.com/user-attachments/assets/9c5d9994-cc33-4534-b742-c93abadb14bb)

### How to reproduce

- Use v2.3.0b3 with the new sonos provider and a Sonos speaker
- Try to make an announcement using the `media_player.play_media` OR `mass.play_announcement` action/service call. Example:

```yaml
action: media_player.play_media
target:
  entity_id: media_player.office_speaker
data:
  announce: true
  media_content_id: |
    media-source://tts/cloud?message=test
  media_content_type: provider
  extra:
    volume: 30
```

### Music Providers

Home assistant cloud tts

### Player Providers

Sonos (S2 speakers)

### Full log output

Ignore `Error handling message: players/cmd/play_announcement: Only URLs are supported for announcements` that was me trying to use the advanced announcement action/service with the wrong url structure which is not related to the correct usage of `media_player.play_media` or `mass.play_announcement`.
[music-assistant.log](https://github.com/user-attachments/files/16738558/music-assistant.log)

### What version of Home Assistant Core are your running

2024.8.2

### What type of installation are you running?

Home Assistant OS

### On what type of hardware are you running?

Generic x86-64 (e.g. Intel NUC)

**Comment by OzGav** [comment]
Can you also try this the mass play announcement service call

**Comment by jordandrako** [comment]
> Can you also try this the mass play announcement service call

Same error when using either announcement method.
![image](https://github.com/user-attachments/assets/da2b75c6-55a3-40c5-9277-3151adb1adf0)

**Comment by marcelveldt** [comment]
I have just published 2.3.0 beta 5 - can you give that a try ?

**Comment by jordandrako** [comment]
Confirmed it works for me now, thank you!
